src/main/config.cpp

```diff
@@ -258,6 +260,25 @@ void DBConfig::SetDefaultMaxMemory() {
 	}
 }
 
+void DBConfig::SetDefaultMaxSwapSpace(optional_ptr<DatabaseInstance> db) {
+	options.maximum_swap_space.SetDefault(0);
+	if (options.temporary_directory.empty()) {
+		return;
+	}
+	if (!db) {
+		return;
+	}
+	auto &fs = FileSystem::GetFileSystem(*db);
+	if (!fs.DirectoryExists(options.temporary_directory)) {
+		// Directory doesnt exist yet, we will look up the disk space once we have created the directory
+		// FIXME: do we want to proactively create the directory instead ???
```

Comment:
This would just make my life easier, but other than that there's no real reason to do this
Though now we are modifying the config during execution (when the temp directory gets created) which might not be something we want to do?